src/aspire/denoising/adaptive_support.py

```diff
@@ -0,0 +1,107 @@
+import logging
+
+import numpy as np
+
+from aspire.image import Image
+from aspire.numeric import fft
+from aspire.utils.coor_trans import grid_2d
+
+logger = logging.getLogger(__name__)
+
+
+def adaptive_support(images, energy_threshold=0.99):
```

Comment:
In the future, it might make more sense for this to take an `ImageSource` as input, since we can't always rely on all images being in memory.

Reply:
I've naively changed to using an `ImageSource` or subclass.

I have not changed the code to be smart (batch-wise calculations) with the image stack internally.  I think this is a fair compromise to get the interface using the class without investing a lot into code that may not get used near term.

Reply:
Makes sense.